presto-product-tests/src/main/java/com/facebook/presto/tests/hive/TestHiveBasicTableStatistics.java

```diff
@@ -145,6 +145,45 @@ public void testCreatePartitioned()
         }
     }
 
+    @Test(groups = {HIVE_CONNECTOR, HIVE_TABLE_STATISTICS})
+    public void testAnalyze()
+    {
+        String tableName = "test_basic_statistics_analyze";
+
+        onPresto().executeQuery(format("DROP TABLE IF EXISTS %s", tableName));
+        onPresto().executeQuery(format("" +
+                "CREATE TABLE %s " +
+                "WITH ( " +
+                "   partitioned_by = ARRAY['n_regionkey'], " +
+                "   bucketed_by = ARRAY['n_nationkey'], " +
+                "   bucket_count = 10 " +
+                ") " +
+                "AS " +
+                "SELECT n_nationkey, n_name, n_comment, n_regionkey " +
+                "FROM nation " +
+                "WHERE n_regionkey = 1", tableName));
+
+        try {
+            BasicStatistics tableStatistics = getBasicStatisticsForTable(onHive(), tableName);
+            assertThatStatisticsAreNotPresent(tableStatistics);
+
+            BasicStatistics partitionStatisticsBefore = getBasicStatisticsForPartition(onHive(), tableName, "n_regionkey=1");
+
+            // run ANALYZE
+            onPresto().executeQuery(format("ANALYZE %s", tableName));
+            BasicStatistics partitionStatisticsAfter = getBasicStatisticsForPartition(onHive(), tableName, "n_regionkey=1");
+
+            // ANALYZE must not change the basic stats
+            assertThat(partitionStatisticsBefore.getNumRows().getAsLong()).isEqualTo(partitionStatisticsAfter.getNumRows().getAsLong());
+            assertThat(partitionStatisticsBefore.getNumFiles().getAsLong()).isEqualTo(partitionStatisticsAfter.getNumFiles().getAsLong());
+            assertThat(partitionStatisticsBefore.getRawDataSize().getAsLong()).isEqualTo(partitionStatisticsAfter.getRawDataSize().getAsLong());
+            assertThat(partitionStatisticsBefore.getTotalSize().getAsLong()).isEqualTo(partitionStatisticsAfter.getTotalSize().getAsLong());
+        }
+        finally {
```

Comment:
no need for `finally` (even if you want to re-run test in a loop, you have DROP IF EXISTS at the beginning)

Reply:
The production test is not a stateless environment and I think it will be better that we clean up the side effect of a failed test right after its run.